packages/zowe-explorer/src/trees/uss/UssFSProvider.ts

```diff
@@ -161,6 +163,22 @@ export class UssFSProvider extends BaseProvider implements vscode.FileSystemProv
         if (FsAbstractUtils.isFileEntry(entry)) {
             return entry;
         }
+        if (entry == null && resp?.success) {
+            // if entry is null, listFiles did not create a new directory entry - this is a file
+            let parentDir = this._lookupParentDirectory(uri, true);
+            if (parentDir == null) {
+                const parentPath = path.posix.join(uri.path, "..");
+                const parentUri = uri.with({ path: parentPath });
+                await vscode.workspace.fs.createDirectory(parentUri);
+                parentDir = this._lookupParentDirectory(uri, false);
```

Comment:
I assume this will throw if the immediate parent (`..`) is not a directory.
And that's why we don't need to `parentDir?.metadata =`

Reply:
Yep that's correct - `createDirectory` will throw if the parent already exists as a file entry.